Cubical/Categories/Site/Instances/ZariskiCommRing.agda

```diff
@@ -105,3 +112,93 @@ pullbackStability zariskiCoverage {c = A} um {d = B} φ =
 
   ψComm : ∀ i → (ψ i .fst) ∘ (AU._/1 (um .f i)) ≡ (BU._/1 (φ $r um .f i)) ∘ φ .fst
   ψComm i = uniqInvElemHom φ (um .f i) .fst .snd
+
+
+-- this defines a subcanonical coverage
+-- the proof is analoguous to the case of lattice sheaves,
+-- see "isLimConeLocCone" in Cubical.Algebra.CommRing.Localisation.Limit
+-- first, a lemma
+module SubcanonicalLemmas (A R : CommRing ℓ) where
+  open CommRingStr ⦃...⦄
+  open CommIdeal R using (_∈_)
+
+  private instance
+   _ = A .snd
+   _ = R .snd
+
+  module _
+    {n : ℕ}
+    (f : FinVec ⟨ R ⟩ n)
+    (isUniModF : 1r ∈ ⟨ f ⟩[ R ])
+    (fam@(φ , isCompatibleφ) : CompatibleFamily (yo A)
+                                                (str (covers zariskiCoverage R)
+                                                (unimodvec n f isUniModF)))
+    where
+    open InvertingElementsBase R
+    open RingHoms
+    module U i = UniversalProp (f i)
+    module UP i j = UniversalProp (f i · f j)
+
+    private
+      _/1ⁱ : ⟨ R ⟩ → {i : Fin n} →  R[1/ (f i) ]
+      (r /1ⁱ) {i = i} = U._/1 i r
+
+
+    applyEqualizerLemma : ∀ a → ∃![ r ∈ ⟨ R ⟩ ] ∀ i → r /1ⁱ ≡ φ i .fst a
+    applyEqualizerLemma a = equalizerLemma R f isUniModF (λ i → φ i .fst a) path
+      where
+      pathHelper : ∀ i j → χˡ R f i j ∘r (U./1AsCommRingHom i)
+                         ≡ χʳ R f i j ∘r (U./1AsCommRingHom j)
+      pathHelper i j = RingHom≡
+                         (χˡUnique R f i j .fst .snd ∙ sym (χʳUnique R f i j .fst .snd))
+
+      path : ∀ i j → χˡ R f i j .fst (φ i $r a) ≡ χʳ R f i j .fst (φ j $r a)
+      path i j = funExt⁻ (cong fst (isCompatibleφ i j _ _ _ (pathHelper i j))) a
+
+    inducedHom : CommRingHom A R
+    fst inducedHom a = applyEqualizerLemma a .fst .fst
+    snd inducedHom = makeIsRingHom
+                       (cong fst (applyEqualizerLemma 1r .snd (1r , 1Coh)))
+                       (λ x y → cong fst (applyEqualizerLemma (x + y) .snd (_ , +Coh x y)))
+                       (λ x y → cong fst (applyEqualizerLemma (x · y) .snd (_ , ·Coh x y)))
+      where
+      open IsRingHom
+      1Coh : ∀ i → (1r /1ⁱ ≡ φ i .fst 1r)
+      1Coh i = sym (φ i .snd .pres1)
+
+      +Coh : ∀ x y i → ((fst inducedHom x + fst inducedHom y) /1ⁱ ≡ φ i .fst (x + y))
+      +Coh x y i = let instance _ = snd R[1/ f i ]AsCommRing in
+           U./1AsCommRingHom i .snd .pres+ _ _
+        ∙∙ cong₂ _+_ (applyEqualizerLemma x .fst .snd i) (applyEqualizerLemma y .fst .snd i)
+        ∙∙ sym (φ i .snd .pres+ x y)
+
+      ·Coh : ∀ x y i → ((fst inducedHom x · fst inducedHom y) /1ⁱ ≡ φ i .fst (x · y))
+      ·Coh x y i = let instance _ = snd R[1/ f i ]AsCommRing in
+           U./1AsCommRingHom i .snd .pres· _ _
+        ∙∙ cong₂ _·_ (applyEqualizerLemma x .fst .snd i) (applyEqualizerLemma y .fst .snd i)
+        ∙∙ sym (φ i .snd .pres· x y)
```

Comment:
This part should be very similar to what happens when one shows that the forgetful functor from rings to sets reflects limits. But I don't see a quick way to make it reusable, so it is probably best to leave it like it is for now.

We should add a comment pointing out the relation.

Reply:
Added a comment now. Do you think that's enough?

Reply:
Yes, nice comment!